src/librustc_driver/driver.rs

```diff
@@ -669,12 +669,24 @@ pub fn phase_2_configure_and_expand<'a>(sess: &Session,
         // dependent dlls. Note that this uses cfg!(windows) as opposed to
         // targ_cfg because syntax extensions are always loaded for the host
         // compiler, not for the target.
-        let mut _old_path = OsString::new();
+        //
+        // This is somewhat of an inherently racy operation, however, as
+        // multiple threads calling this function could possibly continue
+        // extending PATH far beyond what it should. To solve this for now we
+        // just don't add any new elements to PATH which are already there
+        // within PATH. This is basically a targeted fix at #17360 for rustdoc
+        // which runs rustc in parallel but has been seen (#33844) to cause
+        // problems with PATH becoming too long.
+        let mut old_path = OsString::new();
```

Comment:
Nah by using `if` instead of `#[cfg]` it's the same as doing this on Unix:

```rust
if false {
    let foo = bar;
    something(foo);
}
```

which doesn't emit an unused variable warning.